varnish.go

```diff
@@ -212,6 +223,31 @@ func executeVarnishstat(varnishstatExe string, params ...string) (*bytes.Buffer,
 	return buf, cmd.Run()
 }
 
+// Returns the most recent prefix for 'VBE.reload_' stats. Empty until first reload.
+// 'VBE.reload_2019-08-29T100458' as by varnish_reload_vcl in 4.1+
+// 'VBE.reload_20191014_091124_78599' as by varnishreload in 6+
+func findMostRecentVbeReloadPrefix(countersJSON map[string]interface{}) string {
+	var mostRecentVbeReloadPrefix string
+	for vName, _ := range countersJSON {
+		// Checking only the required ".happy" stat
+		if strings.HasPrefix(vName, vbeReload) && strings.HasSuffix(vName, ".happy") {
+			dotAfterPrefixIndex := vbeReloadLength + strings.Index(vName[vbeReloadLength:], ".")
+			vbeReloadPrefix := vName[:dotAfterPrefixIndex]
+			if strings.Compare(vbeReloadPrefix, mostRecentVbeReloadPrefix) > 0 {
+				mostRecentVbeReloadPrefix = vbeReloadPrefix
+			}
+		}
+	}
+	return mostRecentVbeReloadPrefix
```

Comment:
If there is only one set on `reload_` entries, this function will return the only ones timestamp and that will then be (unintentionally?) ignored in `ScrapeVarnishFrom`. Did you intend this, I assume letting the one go through is ok, if its ok to let one of many through as well.
```suggestion
	mostRecentVbeReloadPrefix  := ""
	numReloadedPrefix := 0	
	for vName, _ := range countersJSON {
		// Checking only the required ".happy" stat
		if strings.HasPrefix(vName, vbeReload) && strings.HasSuffix(vName, ".happy") {
			numReloadedPrefix++
			dotAfterPrefixIndex := vbeReloadLength + strings.Index(vName[vbeReloadLength:], ".")
			vbeReloadPrefix := vName[:dotAfterPrefixIndex]
			if strings.Compare(vbeReloadPrefix, mostRecentVbeReloadPrefix) > 0 {
				mostRecentVbeReloadPrefix = vbeReloadPrefix
			}
		}
	}
	if numReloadedPrefix <= 1  {
		return ""
	}
	return mostRecentVbeReloadPrefix
```

Reply:
> If there is only one set on reload_ entries, this function will return the only ones timestamp

Correct

> and that will then be (unintentionally?) ignored in ScrapeVarnishFrom.

I don't think this is true. `ScrapeVarnishFrom` will use `isOutdatedVbe` to determine that the VBE counters are not, in fact, outdated, so they won't be skipped (this should be covered by the `current VCL version` test case of `Test_IsOutdatedVbe`).

Let me know if I missed something!